src/Processors/Executors/ExecutionThreadContext.h

```diff
@@ -43,6 +43,14 @@ class ExecutionThreadContext
     const bool profile_processors;
     const bool trace_processors;
 
+    /// There is a performance optimization that schedules a task to the current thread, avoiding global task queue.
+    /// Optimization decreases contention on global task queue but may cause starvation.
+    /// See 01104_distributed_numbers_test.sql
```

Comment:
It makes sense to add a test for non-distributed case - https://github.com/ClickHouse/ClickHouse/issues/56256#issuecomment-1812675642